youtube_dl/extractor/nexx.py

```diff
@@ -220,6 +220,13 @@ def p0(p):
 
         return formats
 
+    def _protection_tokens(self, video):
+        protection_data = video.get('protectiondata')
+        if protection_data:
+            tokenDASH = protection_data.get('tokenDASH')
```

Comment:
Breaks. You are not guaranteed `protection_data` is a dict. **Do not** change previous token extraction algo.

Reply:
I'll use `isinstance` then, thanks. I didn't change the previous token extraction, it simply prefers these new tokens.

Reply:
Again: no, do not touch token extraction code. Just eliminate duplication.

Reply:
I eliminated duplication and had to reorganize the token extraction code. I touched the previous token extraction algo, but did not change the functionality.

Reply:
Is this some sort of fat trolling or what?

Reply:
No, it isn't. Is the new commit not good? It no longer contains any duplication.

Reply:
Can you suggest how I am supposed to keep the code DRY **without** at least touching the previous token extractor?